qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1145,7 +1192,10 @@ def cancel(self) -> bool:
         Returns:
             True if all jobs and analysis are successfully cancelled, otherwise false.
         """
-        return self.cancel_jobs() and self.cancel_analysis()
+        # Cancel analysis first since it is queued on jobs, then cancel jobs
+        # otherwise there can be a race issue when analysis starts running
+        # as soon as jobs are cancelled
+        return self.cancel_analysis() and self.cancel_jobs()
```

Comment:
Why are these two called with `and`? I can't think of how there would still be jobs running and `cancel_analysis` would return false (since that usually indicates that an analysis callback is running after all the jobs completed), but it feels like if I am calling `cancel()` I at least want an attempt to cancel any running jobs.

Reply:
Yeah good point, it should evaluate each and then return the `and` of the results. The way its written now it might not evaluate the second clause if the first returns false which is incorrect